hail/src/main/scala/is/hail/lir/X.scala

```diff
@@ -874,8 +874,8 @@ class InsnX(val op: Int, _ti: TypeInfo[_], var lineNumber: Int = 0) extends Valu
       case F2D => DoubleInfo
       case DALOAD => DoubleInfo
       // Boolean
```

Comment:
This comment is wrong for [`i2b`](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.i2b).

[`baload`](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.baload) is a cursed instruction. I think it's actually impossible for us to give the right TypeInfo because it depends on the type of the array argument. It looks like we could have some special case logic in `insn` (where we know the arguments) where we could determine the actually correct type info.

Reply:
Ah I see we can use `children.head`.